source/common/runtime/runtime_features.cc

```diff
@@ -47,6 +47,13 @@ constexpr const char* disallowed_features[] = {
     // Acts as both a test entry for deprecated.proto and a marker for the Envoy
     // deprecation scripts.
     "envoy.deprecated_features.deprecated.proto:is_deprecated_fatal",
+    "envoy.deprecated_features.config_source.proto:UNSUPPORTED_REST_LEGACY",
+    "envoy.deprecated_features.ext_authz.proto:use_alpha",
+    "envoy.deprecated_features.route.proto:enabled",
+    "envoy.deprecated_features.tcp_proxy.proto:deprecated_v1",
+    "envoy.deprecated_features.fault.proto:type",
+    "envoy.deprecated_features.cds.proto:hosts",
```

Comment:
The docs aren't even done on this one IIRC. Should we punt this another release? cc @dio @htuch @envoyproxy/maintainers. (This one concerns me a lot from a complaining perspective and I'm still torn whether we should just leave it alone. Thoughts?)

Reply:
After further discussions with @htuch this morning my current feeling is we should just undeprecate this. IMO it's not worth the pain. Would like to hear what others think.

Reply:
I have no objections for making `hosts` fatal-by-default, but un-deprecating sounds quite appealing, since Istio is still using `hosts` (though, I don't believe there are any blockers for migrating away from it).

Reply:
"envoy.deprecated_features.tcp_proxy.proto:deprecated_v1" is used in Istio.

Reply:
Is there a reasonable replacement for `deprecated_v1` or is it still not ready? If the former, we can move forward with the deprecation per the policy. If not, we should undeprecate this field. cc @PiotrSikora

Reply:
`tcp_proxy.deprecated_v1` is not used in Istio, so we don't have objections removing that.

Having said that, Envoy still doesn't have a proper replacement for this feature, since no one implemented #4457 yet.

Reply:
Thanks for @PiotrSikora. I think in that case we also need to undeprecate `tcp_proxy.deprecated_v1` until we have a complete replacement.